frontend/src/app/auth/auth.service.ts

```diff
@@ -19,5 +19,10 @@ export class AuthService {
     return this.http.post('/api/auth/sign-out', {})
   }
 
+  changePassword(data: {currentPassword: string; newPassword: string; username: string}){
+    
+    return this.http.post('/api/auth/change-password', data).pipe(switchMap(() => this.signIn({username: data.username, password: data.newPassword})))
```

Comment:
ta część: `switchMap(() => this.signIn({username: data.username, password: data.newPassword}))`

powinna być w `ChangePasswordComponent.submit`